packages/apify/src/actor.ts

```diff
@@ -186,11 +186,22 @@ export class Actor<Data extends Dictionary = Dictionary> {
         options = typeof messageOrOptions === 'string' ? { ...options, statusMessage: messageOrOptions } : { ...messageOrOptions, ...options };
         options.exit ??= true;
         options.exitCode ??= EXIT_CODES.SUCCESS;
-        options.timeoutSecs ??= 5;
+        options.timeoutSecs ??= 30;
 
-        this.eventManager.emit(EventType.EXIT, options);
+        // Close the event manager and emit the final PERSIST_STATE event
         await this.eventManager.close();
 
+        // Emit the exit event
+        this.eventManager.emit(EventType.EXIT, options);
+
+        // Wait for all event listeners to be processed
+        log.debug(`Waiting for ${options.timeoutSecs} seconds for all event listeners to complete their execution`);
```

Comment:
```suggestion
        log.debug(`Waiting for all event listeners to complete their execution (with ${options.timeoutSecs} seconds timeout)`);
```

also could we maybe print this only if there are any event listeners? we should know the count, right?

Reply:
Well we always have event listeners attached (for PERSIST_STATE) so I think it's better to just log it no matter what